Update the doc to mention that docker is needed to build the jar

## Issue

As the doc dont mention that a docker daemon is running, by consequence we will get such an error during the build
```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] docker run -v /Users/cmoullia/code/dekorate/dekorate-cli/target/dekorate-cli-1.0-SNAPSHOT-native-image-source-jar:/project:z --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java11 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar dekorate-cli-1.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http --no-server -H:-UseServiceLoaderFeature -H:+StackTrace dekorate-cli-1.0-SNAPSHOT-runner
docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.
See 'docker run --help'.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13.283 s
[INFO] Finished at: 2020-07-01T14:56:58+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.5.0.Final:build (default) on project dekorate-cli: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:358)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.lang.RuntimeException: Image generation failed. Exit code: 125
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:374)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:344)
[ERROR] 	... 12 more
[ERROR] -> [Help 1]
```

**Remark**: Typo error `apprroach` should be `approach`

Can we build the project without the need to create a native jar which is taking times and CPU ? To be documented too

Build is failing even if docker is up and running

```
docker run -v /Users/cmoullia/code/dekorate/dekorate-cli/target/dekorate-cli-1.0-SNAPSHOT-native-image-source-jar:/project:z --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java11 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar dekorate-cli-1.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http --no-server -H:-UseServiceLoaderFeature -H:+StackTrace dekorate-cli-1.0-SNAPSHOT-runner
[dekorate-cli-1.0-SNAPSHOT-runner:24]    classlist:   9,243.78 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]        (cap):   1,225.06 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]        setup:   3,814.47 ms
13:24:05,024 INFO  [org.jbo.threads] JBoss Threads version 3.1.1.Final
[dekorate-cli-1.0-SNAPSHOT-runner:24]   (typeflow): 101,866.33 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]    (objects): 207,552.00 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]   (features):  38,155.03 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]     analysis: 404,044.71 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]     (clinit):   4,612.32 ms
[dekorate-cli-1.0-SNAPSHOT-runner:24]     universe:  77,656.57 ms
Error: Image build request failed with exit status 137
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10:31 min
[INFO] Finished at: 2020-07-01T15:33:12+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.5.0.Final:build (default) on project dekorate-cli: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:358)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.lang.RuntimeException: Image generation failed. Exit code: 1
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:374)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:344)
[ERROR] 	... 12 more
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/.gitignore
+++ b/.gitignore
@@ -2,7 +2,6 @@
 .project
 .classpath
 .settings/
-bin/
 
 # IntelliJ
 .idea

--- a/README.md
+++ b/README.md
@@ -36,34 +36,53 @@ For supported runtimes the tool will be able to extract information like:
 That will be applied to the generated manifests. 
 
 # Known issues
-This tool requires `quarkus-picocli` which is not yet releashed.
-So, at the moment is not possible to build a native binary (it's only possible using quarkus snapshot builds).
+
+Currently, it is not possible to build a native binary (see https://github.com/dekorateio/dekorate-cli/issues/16).
   
 # Building and installing
 
-The easiest apprroach to creating a native binary is:
+Dekorate cli comes in two flavours:
 
-```
-./mvnw package -Pnative -Dquarkus.native.container-build=true
-```
+- Uberjar
+
+- Native binary
+
+## Uberjar
+
+To build the uberjar:
+
+    ./mvnw clean install
 
 The command above will generate `/target/dekorate-cli-1.0-SNAPSHOT-runner`.
-You can copy the generated binary to your `$PATH`. For example:
 
-```
-cp `/target/dekorate-cli-1.0-SNAPSHOT-runner $HOME/bin/dekorate`
-```
+You can then use the script found under the `bin` folder to invoke the uberjar.
+You can copy it to your `$PATH`. For example:
+
+    cp `bin/dekorate $HOME/bin/dekorate`
+    chmod +x $HOME/bin/dekorate
 
 The rest of this document will assume that the tool is available in the `$PATH` as `dekorate`.
+
+## Native binary
+
+The easiest apprroach to creating a native binary is:
+
+    ./mvnw package -Pnative -Dquarkus.native.container-build=true
+
+****Note*: The command above requires access to the docker daemon.
+
+If no docker daemon is not available, then you can try to build the native binary locally using:
+
+    ./mvnw package -Pnative
+
+This will only work provided that you have `GraalVM 19.3.1`. For more details please consult: https://quarkus.io/guides/building-native-image
 
 
 # Usage
 
 ## Generating Kubernetes manifests
 
-```
-dekorate kubernetes
-```
+    dekorate kubernetes
 
 ### Generator output
 
@@ -86,9 +105,7 @@ For nodejs project the output directory is always: `kubernetes`.
 
 To specify the name of the resources inside the generated manifests, you can use the `-n, --name` option:
 
-```
-dekorate kubernetes -n my-app
-```
+    dekorate kubernetes -n my-app
 
 This name, will be used to: 
 - name resources
@@ -97,9 +114,7 @@ This name, will be used to:
 
 In the same spirit you can specify the version using `-v, --ver` option:
 
-```
-dekorate kubernetes -v 1.0.0
-```
+    dekorate kubernetes -v 1.0.0
 
 This version, will be used to: 
 - in the `app.kubernetes.io/version` label
@@ -111,38 +126,28 @@ This version, will be used to:
 
 To add one or more labels to the generated manifests, you can use the `-l, --label` option:
 
-```
-dekorate kubernetes -l foo=bar
-```
+    dekorate kubernetes -l foo=bar
 
 The option can be used multiple times:
 
-```
-dekorate kubernetes -l foo=bar -l bar=baz
-```
+    dekorate kubernetes -l foo=bar -l bar=baz
 
 ### Annotations
 
 To add one or more labels to the generated manifests, you can use the `-a, --annotation` option:
 
-```
-dekorate kubernetes -a foo=bar
-```
+    dekorate kubernetes -a foo=bar
 
 The option can be used multiple times:
 
-```
-dekorate kubernetes -a foo=bar -a bar=baz
-```
+    dekorate kubernetes -a foo=bar -a bar=baz
 
 
 ### Environment variables
 
 To add one or more environment variables to the containers inside the generated manifests, you can use the `-e, --env-var` option:
 
-```
-dekorate kubernetes -e FOO=BAR
-```
+    dekorate kubernetes -e FOO=BAR
 
 In addition to simple key value pairs its also possible to define evironment variables sources using:
 - Secret
@@ -152,34 +157,24 @@ In addition to simple key value pairs its also possible to define evironment var
 
 To add a secret as an environment variable source, you can use the `-es, --env-from-secret` option:
 
-```
-dekorate kubernetes -es my-secret
-```
+    dekorate kubernetes -es my-secret
 
 ### Configmap Envrionment variable sources
 
 To add a configmap as an environment variable source, you can use the `-es, --env-from-configmap` option:
 
-```
-dekorate kubernetes -es my-configmap
-```
+    dekorate kubernetes -es my-configmap
 
 
 ## Generating Openshift manifests
 
-```
-dekorate openshfit
-```
+    dekorate openshfit
 
 
 ## Generating Knative manifests
 
-```
-dekorate knative
-```
+    dekorate knative
 
 ## Generating Tekton manifests
 
-```
-dekorate tekton
-```
+    dekorate tekton

--- a/bin/dekorate
+++ b/bin/dekorate
@@ -1,0 +1,9 @@
+#!/bin/bash
+
+VERSION=`ls ~/.m2/repository/io/dekorate/dekorate-cli/ | grep -v xml | sort -r | uniq | head -n 1`
+
+if [ -z "$VERSION" ]; then
+  echo "Failed to detect dekorate-cli under ${HOME}/.m2/repository/ . Ensure you install the binary locally!"
+fi
+java -jar ~/.m2/repository/io/dekorate/dekorate-cli/${VERSION}/dekorate-cli-${VERSION}-runner.jar $*
+

--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -1,2 +1,3 @@
 quarkus.banner.enabled=false
 quarkus.log.level=ERROR
+quarkus.package.uber-jar=true

